jumpscale/clients/taiga/taiga.py

```diff
@@ -342,23 +342,26 @@ def export_all_issues_details(self, path="/tmp/issues_details.md", with_descript
 
         j.sals.fs.write_file(path=path, data=text)
 
+    def __get_single_project_required_data(self, project):
+        single_project_template = dict()
+        single_project_template["name"] = project.name
+        single_project_template["created_date"] = project.created_date
+        single_project_template["owner"] = project.owner.get("full_name_display")
+        single_project_template["issues"] = project.list_issues()
+        return single_project_template
+
     def __get_project_required_data(self, projects):
-        """Get the required data from the projects 
+        """Get the required data from the projects
         Args:
             projects(list): list of projects
 
         Returns:
             all_projects_template (dic): dict of the data required from project object
 
         """
-        all_projects_template = list()
-        for project in projects:
-            single_project_template = dict()
-            single_project_template["name"] = project.name
-            single_project_template["created_date"] = project.created_date
-            single_project_template["owner"] = project.owner.get("full_name_display")
-            single_project_template["issues"] = project.list_issues()
-            all_projects_template.append(single_project_template)
+        threads = [gevent.spawn(self.__get_single_project_required_data, project) for project in projects]
+        gevent.joinall(threads)
+        all_projects_template = [thread.value for thread in threads]
```

Comment:
what happens in case of errors here?

Reply:
Very good question. It returns None for the failed ones. Should I filter out the failed ones, raise an exception in case any one of them failed, retry for 6 times for all of them to succeed and raise in case all of the tries failed?

Reply:
No need to raise (join should have raise_exceptions or similar flag), you can link greenlet to link_exception and maybe have a log error there in that handling, or use a group and wrap the errors e.g https://gist.github.com/progrium/956006